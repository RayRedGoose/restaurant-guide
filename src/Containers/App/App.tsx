import './App.scss';
import React, { useEffect, useState } from 'react';
import { useDispatch } from 'react-redux';
import { getRestaurants } from '_apiCalls/apiCalls';
import { addRestaurants } from 'redux_utils/actions';

const App: React.FC = () => {
  const dispatch = useDispatch();
  const [ error, setError ] = useState<string>('');
  const [ loaded, setLoaded ] = useState<boolean>(false);
  const fetchRestaurants = async (): Promise<void> => {
    try {
      const restaurants = await getRestaurants();
      dispatch(addRestaurants(restaurants));
      setLoaded(true);
    }
    catch (error) {
      setLoaded(false);
      setError(error);
    }
  };

  const addRestaurantsToStore = (): void => { fetchRestaurants() };

  useEffect(addRestaurantsToStore, []);

  return (
    <div className="app">
      { !loaded && <p>Loading ...</p> }
      { loaded && <p>Loading is ended!</p>}
      { error !== '' && <p>{ error }</p> }
    </div>
  );
}

export default App;
