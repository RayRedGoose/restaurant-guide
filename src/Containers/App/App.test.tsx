import React from "react";
import { shallow } from "enzyme";
import App from "./App";
import mockStore from "assets/ts/test/mockStore";

const mockDispatch = jest.fn();
jest.mock("react-redux", () => ({
  useDispatch: () => mockDispatch,
  useSelector: () => ({
    ...mockStore,
    displayPage: 1,
  }),
}));

describe("App", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
