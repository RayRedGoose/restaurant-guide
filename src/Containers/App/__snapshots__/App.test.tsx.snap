// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Tests for happy result of fetching should match the snapshot while fetch is done 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="app"
      >
        <header
          class="sorting-panel"
        >
          <form
            class="search-form"
          >
            <input
              placeholder="Search"
              type="text"
              value=""
            />
            <input
              type="submit"
              value="Search"
            />
            <button
              class="clean-filters"
            >
              Clean
            </button>
          </form>
          <section
            class="filter"
          >
            <h3>
              all states
            </h3>
          </section>
          <section
            class="filter"
          >
            <h3>
              all genres
            </h3>
          </section>
          <section
            class="filter"
          >
            <h3>
              all attires
            </h3>
          </section>
        </header>
        <main
          class="restaurants-container"
        >
          <section
            class="restaurant-card"
          >
            <header>
              <p
                class="genre"
              >
                Steak
              </p>
              <p
                class="genre"
              >
                American
              </p>
              <p
                class="genre"
              >
                Contemporary
              </p>
              <p
                class="genre"
              >
                Seafood
              </p>
              <p
                class="genre"
              >
                Cafe
              </p>
            </header>
            <section
              class="card-description"
            >
              <h2>
                Old Hickory Steakhouse
              </h2>
            </section>
            <section
              class="card-information"
            >
              <p
                class="location"
              >
                Oxon Hill, MD
              </p>
              <p
                class="phone"
              >
                (301) 965-4000
              </p>
            </section>
          </section>
          <footer>
            <p
              class="pages"
            >
              1 / 3
            </p>
            <button
              id="next-page-btn"
            >
               + 
            </button>
          </footer>
        </main>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="app"
    >
      <header
        class="sorting-panel"
      >
        <form
          class="search-form"
        >
          <input
            placeholder="Search"
            type="text"
            value=""
          />
          <input
            type="submit"
            value="Search"
          />
          <button
            class="clean-filters"
          >
            Clean
          </button>
        </form>
        <section
          class="filter"
        >
          <h3>
            all states
          </h3>
        </section>
        <section
          class="filter"
        >
          <h3>
            all genres
          </h3>
        </section>
        <section
          class="filter"
        >
          <h3>
            all attires
          </h3>
        </section>
      </header>
      <main
        class="restaurants-container"
      >
        <section
          class="restaurant-card"
        >
          <header>
            <p
              class="genre"
            >
              Steak
            </p>
            <p
              class="genre"
            >
              American
            </p>
            <p
              class="genre"
            >
              Contemporary
            </p>
            <p
              class="genre"
            >
              Seafood
            </p>
            <p
              class="genre"
            >
              Cafe
            </p>
          </header>
          <section
            class="card-description"
          >
            <h2>
              Old Hickory Steakhouse
            </h2>
          </section>
          <section
            class="card-information"
          >
            <p
              class="location"
            >
              Oxon Hill, MD
            </p>
            <p
              class="phone"
            >
              (301) 965-4000
            </p>
          </section>
        </section>
        <footer>
          <p
            class="pages"
          >
            1 / 3
          </p>
          <button
            id="next-page-btn"
          >
             + 
          </button>
        </footer>
      </main>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`App component Tests for sad result of fetching should match the snapshot while fetch is failed 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="app"
      >
        <header
          class="sorting-panel"
        >
          <form
            class="search-form"
          >
            <input
              placeholder="Search"
              type="text"
              value=""
            />
            <input
              type="submit"
              value="Search"
            />
            <button
              class="clean-filters"
            >
              Clean
            </button>
          </form>
          <section
            class="filter"
          >
            <h3>
              all states
            </h3>
          </section>
          <section
            class="filter"
          >
            <h3>
              all genres
            </h3>
          </section>
          <section
            class="filter"
          >
            <h3>
              all attires
            </h3>
          </section>
        </header>
        <section
          class="error-card"
        >
          <img
            alt="error"
            src="error.svg"
          />
          <h2>
            Fetch failed!
          </h2>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="app"
    >
      <header
        class="sorting-panel"
      >
        <form
          class="search-form"
        >
          <input
            placeholder="Search"
            type="text"
            value=""
          />
          <input
            type="submit"
            value="Search"
          />
          <button
            class="clean-filters"
          >
            Clean
          </button>
        </form>
        <section
          class="filter"
        >
          <h3>
            all states
          </h3>
        </section>
        <section
          class="filter"
        >
          <h3>
            all genres
          </h3>
        </section>
        <section
          class="filter"
        >
          <h3>
            all attires
          </h3>
        </section>
      </header>
      <section
        class="error-card"
      >
        <img
          alt="error"
          src="error.svg"
        />
        <h2>
          Fetch failed!
        </h2>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`App component should match the snapshot while fetch is loading 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="app"
      >
        <header
          class="sorting-panel"
        >
          <form
            class="search-form"
          >
            <input
              placeholder="Search"
              type="text"
              value=""
            />
            <input
              type="submit"
              value="Search"
            />
            <button
              class="clean-filters"
            >
              Clean
            </button>
          </form>
          <section
            class="filter"
          >
            <h3>
              all states
            </h3>
          </section>
          <section
            class="filter"
          >
            <h3>
              all genres
            </h3>
          </section>
          <section
            class="filter"
          >
            <h3>
              all attires
            </h3>
          </section>
        </header>
        <section
          class="loading"
        >
          <img
            alt="loading"
            src="loading.svg"
          />
          <h2>
            Loading...
          </h2>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="app"
    >
      <header
        class="sorting-panel"
      >
        <form
          class="search-form"
        >
          <input
            placeholder="Search"
            type="text"
            value=""
          />
          <input
            type="submit"
            value="Search"
          />
          <button
            class="clean-filters"
          >
            Clean
          </button>
        </form>
        <section
          class="filter"
        >
          <h3>
            all states
          </h3>
        </section>
        <section
          class="filter"
        >
          <h3>
            all genres
          </h3>
        </section>
        <section
          class="filter"
        >
          <h3>
            all attires
          </h3>
        </section>
      </header>
      <section
        class="loading"
      >
        <img
          alt="loading"
          src="loading.svg"
        />
        <h2>
          Loading...
        </h2>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
